src/modules/ekf2/EKF/gps_checks.cpp

```diff
@@ -94,8 +94,7 @@ bool Ekf::collect_gps(const gpsMessage &gps)
 		_mag_strength_gps = get_mag_strength_gauss(lat, lon);
 
 		// request a reset of the yaw using the new declination
-		if ((_params.mag_fusion_type != MagFuseType::NONE)
-		     && !declination_was_valid) {
+		if (!declination_was_valid && _control_status.flags.mag_hdg) {
```

Comment:
Why not if we're in 3D fusion? Maybe we should just request a yaw reset regardless of the fusion type? Or we could move the check in the mag control logic: "mag declination suddenly changed -> reset"

Reply:
Once `mag_aligned_in_flight` and potentially entering mag_3d we'll be getting the declination straight from _state.mag_I.

Anyway, the goal here is to try and minimize all these random scattered latching mag reset requests, so yes let's just push the logic into mag control itself.

Reply:
I got rid of this entirely and added logic in mag control to automatically reset if the declination changes by more than 1 degree (from last mag heading usage).